docker/tarfile/doc.go

```diff
@@ -0,0 +1,3 @@
+// Package tarfile is an internal implementation detail of some transports.
+// Do not use outside of the github.com/containers/image repo!
+package tarfile
```

Comment:
… so, I'd suggest moving the `docker-archive:` transport into a separate `containers/image/docker/archive` package.

(`docker:`, i.e. `docker/docker_image_*`, should also have been a separate subpackage, and eventually we’ll fix that).